src/cps/search.cpp

```diff
@@ -309,13 +309,30 @@ namespace cps::search {
             }
         }
 
-        fs::path calculate_prefix(const fs::path & path) {
+        fs::path calculate_prefix(const std::optional<std::string> & path, const std::string_view filename) {
             // TODO: Windows
             // TODO: /cps/<name-like>
-            std::vector<std::string> split = utils::split(std::string{path}, "/");
-            if (split.back() == "") {
-                split.pop_back();
+            if (path) {
+                fs::path p{path.value()};
+                if (p.stem() == "") {
+                    p = p.parent_path();
+                }
+                auto f = fs::path{filename};
+                if (f.stem() == "") {
+                    f = f.parent_path();
+                }
```

Comment:
Possible off-by-one error? The `cps_path` is the *directory* in which the `.cps` file resides (both per the spec, and in your example), but I think `filename` here is the location of the `.cps` file? I think you need to explicitly always do one iteration of `f = f.parent_path();` before you start comparing.

In fact, if it's the file name, it should never have an empty `.stem()`, correct? In which case, just removing the `if` seems like TRTTD.